src/main/content/game/pool-cdn.ts

```diff
@@ -0,0 +1,103 @@
+import type { DownloadInfo } from "@main/content/downloads";
+import { Downloader } from "@main/content/abstract-content";
+import { removeFromArray } from "$/jaz-ts-utils/object";
+import { GameContentAPI } from "@main/content/game/game-content";
+import { extract7z } from "@main/utils/extract-7z";
+
+const log = logger("pool-cdn.ts");
+
+export class PoolCdnDownloader extends Downloader {
+    cdnUrl = "https://pool-init.beyondallreason.dev";
+    poolDataUrl = `${this.cdnUrl}/data.7z`;
+
+    constructor(gameApi: GameContentAPI) {
+        super();
+        this.onDownloadStart.add((downloadInfo) => {
+            gameApi.onDownloadStart.dispatch(downloadInfo);
+        });
+        this.onDownloadComplete.add((downloadInfo) => {
+            gameApi.onDownloadComplete.dispatch(downloadInfo);
+        });
+        this.onDownloadProgress.add((downloadInfo) => {
+            gameApi.onDownloadProgress.dispatch(downloadInfo);
+        });
+        this.onDownloadFail.add((downloadInfo) => {
+            gameApi.onDownloadFail.dispatch(downloadInfo);
+        });
+    }
+
+    /**
+     * Download and extract pool data from the pool CDN, if pool directory does not exist in content folder.
+     *
+     * Will try to reuse existing download if it exists (DownloadHelper will resume download).
+     */
+    public async preloadPoolData() {
+        const poolDirPath = path.join(CONTENT_PATH, "pool");
+        try {
+            await fs.promises.stat(poolDirPath);
+            log.debug("Pool folder exists, skipping download");
+            return;
+        } catch (e) {
+            if ((e as NodeJS.ErrnoException).code !== "ENOENT") {
+                console.error("Error checking pool folder", e);
+                throw e;
+            }
+            // Pool folder does not exist, create and continue
+            await fs.promises.mkdir(poolDirPath);
+        }
+        log.info("Pool folder does not exist, downloading pool data");
+
+        const downloadInfo: DownloadInfo = {
+            type: "game",
+            name: "pool-data",
+            currentBytes: 0,
+            totalBytes: 1,
+        };
+        this.currentDownloads.push(downloadInfo);
+
+        const dlFilePath = path.join(CONTENT_PATH, "data.7z");
+        const dl = new DownloaderHelper(this.poolDataUrl, CONTENT_PATH, {
+            fileName: "data.7z",
+            override: true,
+            resumeIfFileExists: true,
+            resumeOnIncomplete: true,
+            removeOnStop: false,
+            removeOnFail: false,
+        });
+        dl.on("start", () => {
+            log.info("Pool data download started");
+            this.downloadStarted(downloadInfo);
+        });
+        dl.on("end", () => {
+            log.info("Pool data download complete");
+            this.downloadComplete(downloadInfo);
+        });
+        dl.on("error", (error) => {
+            console.error("Pool data download failed", error);
+            this.downloadFailed(downloadInfo);
```

Comment:
This doesn't fully fulfill the

> If the download fails: that's fine, the lobby should ignore and fallback to the regular pr-downloader download.

from the issue

Reply:
It's not a big thing, fine to keep it as is, until people start to complain.

Reply:
Is there a special handler that "stops" the whole lobby in case something calls `donwloadFailed` ?
I've probably not noticed it then, as my understanding was that it would flash the error for a bit but because no error is propagated out and nothing is stopped, the usual process of initial-setup (pr-downloader) continues anyway.

Reply:
Oh, ok, you are right. I've misread it, I've tested now and works as intended, sorry!!!

> Edit: Does this also mean, pool cdn download should have replaced the pr-downloader step for maps? And only fallback to pr-d if pool fails?

I don't follow this question, pool is only for game files, it has nothing to do with maps.

Reply:
> I don't follow this question, pool is only for game files, it has nothing to do with maps.

Typo, I meant the game files, and no worries, more eyes means better code:)

Reply:
Ah, then the answer is no, pr-downloader must be always called. `pool` directory is only "raw content", updated only once a week, and pr-downloader still need to download "index files" that point into this content and fetch the small number of missing files.